haystack/document_store/elasticsearch.py

```diff
@@ -754,13 +758,26 @@ def describe_documents(self, index=None):
                  }
         return stats
 
-    def update_embeddings(self, retriever, index: Optional[str] = None, batch_size: int = 10_000):
+    def update_embeddings(
+        self,
+        retriever,
+        index: Optional[str] = None,
+        filters: Optional[Dict[str, List[str]]] = None,
+        update_existing_embeddings: bool = True,
+        batch_size: int = 10_000
+    ):
         """
         Updates the embeddings in the the document store using the encoding model specified in the retriever.
         This can be useful if want to add or change the embeddings for your documents (e.g. after changing the retriever config).
 
         :param retriever: Retriever to use to update the embeddings.
         :param index: Index name to update
+        :param update_existing_embeddings: Whether to update existing embeddings of the documents. If set to False,
```

Comment:
We can add `embedded_at` field in store. So `update_existing_embeddings` is `False` then it will update embedding for docs where  `embedded_at is None or updated_at > embedded_at`. This will cover the case when user just updated the text of the document but forgot to update embedding.

Reply:
Hi @lalitpagaria, that's a good idea, but it'd be tricky to get it consistent across different document stores. For instance, in the current implementation of FAISS, the vectors cannot be updated. This also brings up an interesting point if we should consider tighter integration between Document Stores & Retrievers to avoid similar inconsistencies.

Reply:
Actually FAISS composite indexes can be used in association with `IDMap`. Which support `remove_ids` method. https://github.com/facebookresearch/faiss/blob/28edc56fa890c8a562f9083b13aeb8dfd594ea50/tests/test_index_composite.py#L53
I suggested @tholor this idea in the past to add `IDMap` before any user supplied index. With that we can achieve remove_id support.

> This also brings up an interesting point if we should consider tighter integration between Document Stores & Retrievers to avoid similar inconsistencies.

Actually in my view embedding logic should reside with DocumentStore, they should support all life cycle of document with embedding as well. And retriever only take care of searching in it. In this way we can decouple embedding from DPR itself. Because there are many models can be used to embedd the text